src/lib/y2_alternatives/dialog/list_alternatives.rb

```diff
@@ -104,10 +104,14 @@ def update_table(index)
       end
 
       def accept_handler
-        @alternatives_list.each(&:save)
+        @alternatives_list.each(&:save) if permissions?
         finish_dialog
       end
 
+      def permissions?
```

Comment:
I also wonder if this logic belongs to the dialog. We have other options:

- The rails way: `Alternative#save` runs all checks and handles the possible exceptions coming from `Y2Alternatives::Control` returning false if something was wrong (or true otherwise).
- Using exceptions: `Alternative#save` checks permissions and raises an exception if needed.
- ...

In both cases, checking `Process::UID` will be done in the `Alternative` class and not in the dialog. It does not belong here.